### ISSUE TITLE
Initial Setup

### ISSUE THREAD

**Issue by skipjack** [webpack-contrib/voting-app#2]
It works (`npm start`)!

What still needs to be done:

- [x] `webpack.dist.js` is not fully fleshed out yet... we'll need to tweak things a bit in order to easily consume from the main repo.
- [x] review and tweak the rest of the configuration, I started it out with what I think is a good boilerplate for a react project but we can tweak it however.
- [x] refactoring, especially in the `Wrapper.jsx` which essentially contains the whole hierarchy. This has a huge `render` method which contains a lot of room for improvement (e.g. breaking a lot out into separate components, refactoring and removing inline callback functions, etc.)

Some nice potential additions that may be helpful going forward:

- maybe react-hot-loader for a nicer development workflow
- use automatic changelog generation and releasing as outlined in [this pr](https://github.com/webpack/webpack.js.org/pull/1017)
- use postcss/cssnext and no sass? CSSNext contains many of the features sass does and has the benefit of actually forcing us to be compliant with what will be supported soon. Plus we're already using postcss for the auto-prefixing so this would allow for just a simpler configuration in general.

I'll create a separate issue for the redesign but I think we should do a little more cleanup first. @NejcZdovc @pksjce or anyone else interested in helping to refactor, please just let me know what you want to tackle and I'll leave that section alone.

**Comment by skipjack** [comment]
Ok, once the two PRs (#6 and #5) are in we can probably close this unless someone else has other configuration/setup ideas for the short-term. Otherwise, things like react-hot-loader, changelog/release automation, and moving to just postcss can all be thought out in other issues and prs.

I have a `port-voting-app` branch started in webpack.js.org to refactor things on that side once this is ready.

**Comment by skipjack** [comment]
@pksjce @NejcZdovc i'll leave merging those PRs to you both, let me know if there's anything else I can do to assist. Otherwise, I'll help integrate in webpack.js.org once the first release has been publish to npm.
